Source/JavaScriptCore/bytecompiler/BytecodeGenerator.h

```diff
@@ -762,7 +762,7 @@ namespace JSC {
         }
 
         RegisterID* emitPutByVal(RegisterID* base, RegisterID* property, RegisterID* value);
-        RegisterID* emitPutByVal(RegisterID* base, RegisterID* thisValue, RegisterID* property, RegisterID* value);
+        RegisterID* emitPutByVal(RegisterID* base, RegisterID* thisValue, RegisterID* property, RegisterID* value, std::optional<ECMAMode> = std::nullopt);
```

Comment:
Let's always have that parameter explicitly (and not use `std::optional`). It is hard to see whether this is missing or not from the code reading. Let's call `ecmaMode()` in the caller of this function

Reply:
Since there are 8 other call sites that would require adding `generator.ecmaMode()` argument, I've introduced instead `emitPutByValWithECMAMode` method to avoid any possible confusion. I hope this resolves your concern.